tests/bsim/bluetooth/host/adv/resume2/dut/main.c

```diff
@@ -0,0 +1,212 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/atomic_builtin.h>
+#include <zephyr/sys/atomic_types.h>
+
+#include <testlib/conn.h>
+#include <testlib/scan.h>
+
+#include <bs_tracing.h>
+#include <bstests.h>
+
+extern enum bst_result_t bst_result;
+
+LOG_MODULE_REGISTER(dut, LOG_LEVEL_INF);
+
+atomic_t connected_count;
+
+static void on_connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	atomic_t count = atomic_inc(&connected_count) + 1;
+
+	LOG_INF("Connected. Current count %d", count);
+}
+
+static void on_disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	atomic_t count = atomic_dec(&connected_count) - 1;
+
+	LOG_INF("Disconnected. Current count %d", count);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = on_connected,
+	.disconnected = on_disconnected,
+};
+
+static void disconnect_all(void)
+{
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+		int err;
+		struct bt_conn *conn = bt_testlib_conn_unindex(BT_CONN_TYPE_LE, i);
+
+		if (conn) {
+			err = bt_testlib_disconnect(&conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+			__ASSERT_NO_MSG(!err);
+		}
+	}
+}
+
+/* 📜:
+ *   🚧Setup
+ *   ✨Setup / Cleanup ok
+ *   👉Test step
+ *   ✅Test step passed
+ *   🚩Likely triggers problematic behavior
+ *   💣Checks for the bad behavior
+ *   💥Bad behavior
+ *   🧹Clean up
+ *   🌈Test complete
+ */
+
+int main(void)
+{
+	int err;
+	bt_addr_le_t connectable_addr;
+	struct bt_conn *conn = NULL;
+
+	bst_result = In_progress;
+
+	err = bt_enable(NULL);
+	__ASSERT_NO_MSG(!err);
+
+	err = bt_set_name("dut");
+	__ASSERT_NO_MSG(!err);
+
+
+	LOG_INF("👉 Preflight test: Advertiser fills connection capacity.");
```

Comment:
Guys this is bonkers... this file is full of emojis... I would have expected we would only accept ascii chars in source.

Reply:
😮

Reply:
More context after talking with @aescolar : 
- we have blind contributors, their accessibility SW/HW doesn't support 
- some greybeard contributors use editors w/ no emoji support (or partial, like eclipse)

I'm reworking the bsim docs, I will add a point for that in the "Conventions" section.